lib/findPort.js

```diff
@@ -35,7 +35,10 @@ module.exports = function(host, start, endOrCallback, callback) {
 	});
 
 	function isInteger(port, cb) {
-		cb(Number.isInteger(port) && port > 0);
+		if (typeof port == 'number')
```

Comment:
full if statement if possible please

Reply:
Also would be nice to get rid of sync callbacks, maybe use `Array.filter` instead of `async.filter`?

Reply:
if statement: Seemed a little too verbose and introduced duplications.  It's suppose to fail if the port is a not a number > 0 or if it's not a number. If full if statement, you'd be calling cb(true) or cb(false). Maybe sticking the bool in a variable then calling cb would be a good alternative?

```js
var portIsIntegerAndGtZero = Number.isInteger(port) && port > 0;
cb(portIsIntegerAndGtZero);
```

otherwise....
```js
if(Number.isInteger(port) && port > 0) {
  cb(true);
} else {
  cb(false);
}
```
filter: Didn't your code do port testing in parallel? I guess calling async.filter is kind of meaningless since we're dealing with an already tested array of values and we just want to know if they are integers > 0.